### PR TITLE
issue w/ plugins detective

### PR DESCRIPTION
**Pull request by mattiaerre**
# Description

as per #445 this functionality is broken; this PR tries to fix the `parse` logic. This, unfortunately, will not fix the fact that the Registry still will not throw any exception for the missing plugin(s); have a look at the issue mentioned above for [more details](https://github.com/opentable/oc/issues/445#issuecomment-304744155).

### CODE REVIEW

**Comment by mattiaerre** [comment]
@matteofigus ... 😊

File: test/unit/code.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = "module.exports=function(a){function b(d){if(c[d])return c[d].exports;var e=c[d]={i:d,l:!1,exports:{}};return a[d].call(e.exports,e,e.exports,b),e.l=!0,e.exports}var c={};return b.m=a,b.c=c,b.i=function(d){return d},b.d=function(d,e,f){b.o(d,e)||Object.defineProperty(d,e,{configurable:!1,enumerable:!0,get:f})},b.n=function(d){var e=d&&d.__esModule?function(){return d['default']}:function(){return d};return b.d(e,'a',e),e},b.o=function(d,e){return Object.prototype.hasOwnProperty.call(d,e)},b.p='',b(b.s=0)}([function(a,b){'use strict';Object.defineProperty(b,'__esModule',{value:!0});b.data=function(f,g){var h={logo:{href:'',url:f.params.logoUrl,alt:f.params.title+' logo'},title:f.params.title,timezoneOffset:'?'};f.plugins.getTimezoneOffset&&(h.timezoneOffset=f.plugins.getTimezoneOffset()),g(null,h)}}]);";
```
Comment (matteofigus) [inline test/unit/code.js]: Can we move this file to `/test/fixtures`? I think that's supposed to be the place for this kind of files.

Comment (mattiaerre) [inline test/unit/code.js]: sure, although I like to keep related things together 😊

Comment (matteofigus) [inline test/unit/code.js]: I think my preference would be to make it a component and bring it to `/test/fixtures/mocked-components/missing-plugin.js` (look at the others in that dir to follow same convention)

Comment (mattiaerre) [inline test/unit/code.js]: @matteofigus I will try to do that although what I do actually need here is only generated code, but I see your point

File: test/unit/registry-domain-plugins-detective.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 const expect = require('chai').expect;
+const code = require('../fixtures/mocked-components/package-server-with-plugin');
```
Comment (mattiaerre) [inline test/unit/registry-domain-plugins-detective.js]: @matteofigus what do you think about this? I only added the `data` part and not the full object; you know that I'm a minimalist 😅

**Pull request merged**